Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/ui/consent/one/DccTicketingConsentOneViewModel.kt

```diff
@@ -1,31 +1,138 @@
 package de.rki.coronawarnapp.dccticketing.ui.consent.one
 
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.asLiveData
 import dagger.assisted.Assisted
 import dagger.assisted.AssistedFactory
 import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.dccticketing.core.common.DccTicketingException
+import de.rki.coronawarnapp.dccticketing.core.service.DccTicketingRequestService
 import de.rki.coronawarnapp.dccticketing.core.transaction.DccTicketingTransactionContext
+import de.rki.coronawarnapp.dccticketing.ui.shared.DccTicketingSharedViewModel
 import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.encryption.ec.EcKeyGenerator
 import de.rki.coronawarnapp.util.ui.SingleLiveEvent
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
 import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import timber.log.Timber
 
 class DccTicketingConsentOneViewModel @AssistedInject constructor(
-    @Assisted private val dccTicketingTransactionContext: DccTicketingTransactionContext,
-    dispatcherProvider: DispatcherProvider
+    @Assisted private val dccTicketingSharedViewModel: DccTicketingSharedViewModel,
+    dispatcherProvider: DispatcherProvider,
+    private val dccTicketingRequestService: DccTicketingRequestService,
+    private val ecKeyGenerator: EcKeyGenerator
 ) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
 
-    val showCloseDialog = SingleLiveEvent<Unit>()
+    private val currentIsLoading = MutableStateFlow(false)
+    val isLoading: LiveData<Boolean> = currentIsLoading.asLiveData()
 
-    val uiState = MutableStateFlow(UiState(dccTicketingTransactionContext)).asLiveData()
+    private val currentEvent = SingleLiveEvent<DccTicketingConsentOneEvent>()
+    val events = currentEvent
 
-    fun goBack() {
-        showCloseDialog.postValue(Unit)
+    private val currentUiState = dccTicketingSharedViewModel.transactionContext
+        .map { UiState(it) }
+    val uiState: LiveData<UiState> = currentUiState.asLiveData2()
+
+    fun onUserCancel() {
+        Timber.d("onUserCancel()")
+        postEvent(ShowCancelConfirmationDialog)
+    }
+
+    fun goBack() = postEvent(NavigateBack)
+
+    fun showPrivacyInformation() = postEvent(NavigateToPrivacyInformation)
+
+    fun onUserConsent(): Unit = launch {
+        Timber.d("onUserConsent()")
+        currentIsLoading.compareAndSet(expect = false, update = true)
+        processUserConsent()
+        currentIsLoading.compareAndSet(expect = true, update = false)
+    }
+
+    private suspend fun processUserConsent() = try {
+        Timber.d("processUserConsent()")
```

Comment:
Can you move all logic to a separate class, so we can easily setup integration tests without the UI?

Reply:
adjusted